Initial terraform cleanup and reorganisation

eternaltyro opened the pull request.

eternaltyro committed:
Initial terraform cleanup and reorganisation

dakotabenjamin commented on deployment/osm-stats.tf:
```diff
@@ -1,86 +1,27 @@
 }
 
 resource "azurerm_container_group" "osm-stats" {
   # this is created within a container group rather than App Service because it
```
very minor but this comment got cut off

eternaltyro replied:
Sorted.

dakotabenjamin commented on deployment/osm-stats.tf:
```diff
@@ -89,288 +30,152 @@ resource "azurerm_container_group" "osm-stats" {
+        azurerm_postgresql_server.osm-stats.fqdn,
+        var.db_name
+      ) **/
+      OVERPASS_URL = "http://export.hotosm.org:6080"
```
Can you help me understand these variables? Why is the OVERPASS_URL hardcoded?

eternaltyro replied:
It shouldn't be. I parameterized it.

dakotabenjamin commented on deployment/osm-stats.tf:
```diff
@@ -89,288 +30,152 @@ resource "azurerm_container_group" "osm-stats" {
-  name = "osm-stats-api-template"
-  resource_group_name = "${azurerm_resource_group.osm-stats.name}"
-
-  template_body = <<DEPLOY
```
Great to see this issue got resolved

eternaltyro replied:
Still needs testing.

dakotabenjamin reviewed (commented):
Just did a once-over of the code, it all looks good to me! Shall we test it out on our azure resources?

eternaltyro committed:
Incorporate customization and parameterise Docker image URLs

eternaltyro committed:
Remove fluff

eternaltyro committed:
Remove backup file

eternaltyro committed:
Update provider

- Update provider block with later version of terraform
- Merge backend block into provider config

eternaltyro committed:
Fix provider authentication

eternaltyro committed:
Fix variable names  for docker images

eternaltyro committed:
Fix errors and make output sensitivity paremeters explicit

eternaltyro committed:
Fix sensitivity for outputs

eternaltyro committed:
Make PostgreSQL publicly available

eternaltyro committed:
Fix duplicate name for web app
